pkg/reconciler/brokercell/resources/deployments.go

```diff
@@ -135,6 +135,13 @@ func deploymentTemplate(args Args, containers []corev1.Container) *appsv1.Deploy
 		},
 		Spec: appsv1.DeploymentSpec{
 			Selector: &metav1.LabelSelector{MatchLabels: Labels(args.BrokerCell.Name, args.ComponentName)},
+			Strategy: appsv1.DeploymentStrategy{
+				RollingUpdate: &appsv1.RollingUpdateDeployment{
+					MaxSurge:       &intstr.IntOrString{IntVal: 1},
+					MaxUnavailable: &intstr.IntOrString{IntVal: 0},
+				},
+			},
+			MinReadySeconds: 60,
```

Comment:
Interesting investigation on the task page!
Regarding MinReadySeconds, I wonder if this applies to upgrade only or does it also apply to new replicas added during autoscaling?

Reply:
It should impact that too.

Reply:
I see. And there is no such option that is upgrade-specific? It's just in the context of our recent work on fanout where we rely increasingly more on autoscaling, I was worrying if an additional the scale out delay of 45 seconds may come at some cost.

Reply:
I don't see any problem at the moment. This delay doesn't stop the pod from being ready and start pulling pubsub.

Reply:
Cool, I have misinterpreted available vs ready then.